lms/djangoapps/grades/signals.py

```diff
@@ -124,19 +125,33 @@ def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=u
     """
     if not settings.FEATURES.get('ENABLE_SUBSECTION_GRADES_SAVED', False):
         return
-    try:
-        course_id = kwargs.get('course_id', None)
-        usage_id = kwargs.get('usage_id', None)
-        student = kwargs.get('user', None)
 
-        course_key = CourseLocator.from_string(course_id)
-        usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    from lms.djangoapps.grades.transformer import GradesTransformer
+    from lms.djangoapps.grades.new.subsection_grade import SubsectionGradeFactory
+    from openedx.core.djangoapps.content.block_structure.api import get_block_structure_manager
```

Comment:
So it looks like the unnecessary coupling that causes this circularity is that the signals and the signal receivers are being bundled together.  If the receivers were put in a different module from the signals, then `module_render.py` would only depend on the signals, which do no importing, not the receivers.